include/ygm/container/oversized_bag.hpp

```diff
@@ -0,0 +1,335 @@
+    /* ---This is the original code---
+    std::swap(m_local_bag, other.m_local_bag);
+    */
+    return *this;
+  }
+
+  using detail::base_async_insert_value<oversized_bag<Item>, for_all_args>::async_insert;
+
+  void async_insert(const Item &value, int dest) {
+    auto inserter = [](auto pcont, const value_type &item) {
+      pcont->local_insert(item);
+    };
+
+    m_comm.async(dest, inserter, this->get_ygm_ptr(), value);
+  }
+
+  /** @todo ----------------------------------------------unchanged----------------------------------------------*/
+  void async_insert(const std::vector<Item> &values, int dest) {
+    auto inserter = [](auto pcont, const std::vector<Item> &values) {
+      for (const auto &v : values) {
+        pcont->local_insert(v);
+      }
+    };
+
+    m_comm.async(dest, inserter, this->get_ygm_ptr(), values);
+  }
+  /** @todo --------------------------------------------end unchanged--------------------------------------------*/
+
+  /** @todo */
+  void local_insert(const Item &val) { /*---This is the original code--- m_local_bag.push_back(val); */ }
+
+  /** @todo */
+  void local_clear() { /*---This is the original code--- m_local_bag.clear(); */ }
+
+  /** @todo */
+  size_t local_size() const { /*---This is the original code--- return m_local_bag.size(); */ }
+
+  /** @todo */
+  size_t local_count(const value_type &val) const {
+    /*---This is the original code--- 
+    return std::count(m_local_bag.begin(), m_local_bag.end(), val);
+    */
+  }
+
+  /** @todo */
+  template <typename Function>
+  void local_for_all(Function fn) {
+    /*---This is the original code---
+    std::for_each(m_local_bag.begin(), m_local_bag.end(), fn);
+    */
+  }
+
+  /** @todo */
+  template <typename Function>
+  void local_for_all(Function fn) const {
+    /*---This is the original code---
+    std::for_each(m_local_bag.cbegin(), m_local_bag.cend(), fn);
+    */
+  }
+  
+  /** @todo */  
+  void serialize(const std::string &fname) {
+    /*---This is the original code---
+    m_comm.barrier();
+    std::string   rank_fname = fname + std::to_string(m_comm.rank());
+    std::ofstream os(rank_fname, std::ios::binary);
+    cereal::JSONOutputArchive oarchive(os);
+    oarchive(m_local_bag, m_comm.size());
+    */
+  }
+
+  /** @todo */  
+  void deserialize(const std::string &fname) {
+    /*---This is the original code---
+    m_comm.barrier();
+
+    std::string   rank_fname = fname + std::to_string(m_comm.rank());
+    std::ifstream is(rank_fname, std::ios::binary);
+
+    cereal::JSONInputArchive iarchive(is);
+    int                      comm_size;
+    // iarchive(m_local_bag, m_round_robin, comm_size);
+    iarchive(m_local_bag, comm_size);
+
+    if (comm_size != m_comm.size()) {
+      m_comm.cerr0(
+          "Attempting to deserialize bag_impl using communicator of "
+          "different size than serialized with");
+    }
+    */
+  }
+
+  /** @todo */  
+  void rebalance() {
```

Comment:
Is rebalancing the bag ensuring that the data corresponding to it is stored in files of similar length?

Reply:
Rebalancing in the sense of the ygm::bag ensures each rank has the same data after a period. We'll look to ensure the same # of total entries are on each rank in our version